test/board_test.rb

```diff
@@ -56,4 +56,23 @@ def test_can_place_ship
   def test_is_occupied
     assert_equal false, @board.is_occupied?(["A1", "A2", "A3"])
   end
+
+  def test_can_place_ship_on_board
+    @board.place(@cruiser, ["A1", "A2", "A3"])
+
+    cell_1 = @board.cells["A1"]
+    cell_2 = @board.cells["A2"]
+    cell_3 = @board.cells["A3"]
+
+    assert_equal @cruiser, cell_1.ship
+    assert_equal @cruiser, cell_2.ship
+    assert_equal @cruiser, cell_3.ship
+    assert_equal true, cell_3.ship == cell_2.ship
+  end
+
+  def test_can_not_overlap_ships_on_board
```

Comment:
Please check if this test is written correctly. It passes but I'm not sure if I'm missing anything.

Reply:
This looks correct to me - You've tested .place, .ship. and .valid_placement and edge cases.  👍